test/e2e/major_upgrade_v3.go

```diff
@@ -97,12 +99,51 @@ func MajorUpgradeToV3(logger *log.Logger) error {
 				testnet.NoError("failed to get header", err)
 				if resp.Header.Version.App == v3.Version {
 					upgradeComplete = true
+					if upgradedHeight == 0 {
+						upgradedHeight = resp.Header.Height
+					}
 				}
 				logger.Printf("height %v", resp.Header.Height)
 				lastHeight = resp.Header.Height
 			}
 		}
 	}
 
+	// check if the timeouts are set correctly
+
+	rpcNode := testNet.Nodes()[0]
+	client, err := rpcNode.Client()
+	testnet.NoError("failed to get client", err)
+	blockTimes := make([]time.Duration, 0, 7)
+	latestBlockTime := time.Time{}
+	nilTime := time.Time{}
+	for h := upgradedHeight - 4; h <= upgradedHeight+4; h++ {
+		resp, err := client.Header(ctx, nil)
+		testnet.NoError("failed to get header", err)
+		if latestBlockTime.Equal(nilTime) {
+			continue
+		}
+
+		blockTime := resp.Header.Time
+		blockDur := blockTime.Sub(latestBlockTime)
+		blockTimes = append(blockTimes, blockDur)
+		latestBlockTime = blockTime
+	}
+
+	if len(blockTimes) < 7 {
+		testnet.NoError("", fmt.Errorf("not enough block times collected: %v", len(blockTimes)))
+	}
+
+	startDur := blockTimes[0]
+	endDur := blockTimes[len(blockTimes)-1]
+
+	if startDur < time.Second*10 {
+		testnet.NoError("", fmt.Errorf("blocks for v2 are too short %v", len(blockTimes)))
+	}
+
+	if endDur > time.Second*7 {
+		testnet.NoError("", fmt.Errorf("blocks for v3 are too long %v", len(blockTimes)))
+	}
```

Comment:
We need to change this to be more robust against flakes by accounting for rounds with more than 1 round.